app/models/piece.rb

```diff
@@ -42,6 +55,12 @@ def obstructed?(destination_x, destination_y)
 
   private
 
+  def remove_piece(destination_x, destination_y)
+    destination_piece = Piece.find_by column_coordinate: destination_x, row_coordinate: destination_y
```

Comment:
Here we can maybe pass the piece we are going to remove from the destination as a parameter instead of doing a query of the pieces table.  Heading in the right direction?